server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -457,20 +429,106 @@ describe("Routerlicious", () => {
 							v2,
 						);
 						const clientId = connectMessage.clientId;
-						const client: Client = {
+						const client: TestSignalClient = {
 							socket,
 							clientId,
-							signalCount: 0,
+							signalsReceived: [],
+							nacksReceived: [],
+							version: v2 ? 2 : 1,
 						};
 
 						return client;
-					};
+					}
 
-					let clientSignalPromises: Promise<ISignalMessage[]>[];
+					function listenForSignals(clients: TestSignalClient[]) {
+						clients.forEach((client) => {
+							client.socket.on("signal", (message: ISignalMessage) => {
+								if (message.clientId !== null) {
+									client.signalsReceived.push(message);
+								}
+							});
+						});
+					}
+
+					function verifyExpectedClientSignals(
+						clients: TestSignalClient[],
+						expectedSignals: ISignalMessage[],
+					) {
+						clients.forEach((client, index) => {
+							assert.equal(
+								client.signalsReceived.length,
+								expectedSignals.length,
+								`User ${index + 1} should have received ${
+									expectedSignals.length
+								} signal(s)`,
+							);
+							expectedSignals.forEach((signal, signalIndex) => {
+								const receivedSignal = client.signalsReceived[signalIndex];
+								Object.keys(signal).forEach((property) => {
+									assert.deepEqual(
+										receivedSignal[property],
+										signal[property],
+										`User ${index + 1} signal ${
+											signalIndex + 1
+										} ${property} mismatch`,
+									);
```

Comment:
Why enumerate properties and not a high level deepEqual?
This enumeration goes over the expected properties but ignores any extra properties that are received. Is that okay? If you are hoping to reveal a specific mismatch, you could dump json of both in the failure message.